### ISSUE TITLE
position: fixed headers overlap document content

### ISSUE THREAD

**Issue by nebulans** [Kozea/WeasyPrint#259]
I'm trying to generate a document with complex headers (these must contain markup, so at present can't be implemented using @page rules) by using the previously suggested here and elsewhere method of making the element position: fixed.  This works to put the header in the document in the appropriate places, but it places it over the content of the document.  I need the content of the document to flow around this header.  How can I achieve this?

See https://gist.github.com/nebulans/633c7911d3d09edfa753 for a sample document that demonstrates this.

**Comment by SimonSapin** [comment]
Try `bottom: 100%` instead of `top: 0`. This places the bottom of your header at a distance from the bottom of the page content area that is 100% of the height of the page content area. That is, at the top of the page content area. Then you can add some `margin-bottom`.

**Comment by nebulans** [comment]
This works after a fashion, but the page content still does not flow around the element.  I've updated the gist, the output then looks like this:
![weasyprint navigator - chromium_021](https://cloud.githubusercontent.com/assets/2145965/8458628/0ebae8d2-2010-11e5-85eb-cc31ed12d0ba.png)

**Comment by SimonSapin** [comment]
I’m not sure what you mean by "flow around the content". If your header is too big for the margin area, you can make the margin area larger with e.g. `@page { margin-top: 5em }`. Same applies to `@page { @top-center { … }}`, by the way. Unfortunately, there is currently no way in WeasyPrint to do this automatically based on the amount of content in the header.

**Comment by liZe** [comment]
@nebulans You'd probably like to see something like `float: top` implemented. It's currently defined in the [CSS Page Floats Working Draft](http://www.w3.org/TR/2015/WD-css-page-floats-3-20150915/#float-property).

**Comment by MichaelBrazier** [comment]
Is there any progress on implementing page floats?

**Comment by liZe** [comment]
> Is there any progress on implementing page floats?

No, but we would be happy to help if someone wants to implement them!
